test/libyul/Parser.cpp

```diff
@@ -804,6 +804,67 @@ BOOST_AUTO_TEST_CASE(customSourceLocations_with_code_snippets_with_nested_locati
 	CHECK_LOCATION(literal128.debugData->location, "source1", 96, 165);
 }
 
+BOOST_AUTO_TEST_CASE(astid)
+{
+	ErrorList errorList;
+	ErrorReporter reporter(errorList);
+	auto const sourceText = R"(
+		/// @src -1:-1:-1 @ast-id 7
+		{
+			/** @ast-id 2 */
+			function f(x) -> y {}
+			mstore(1, 2)
+		}
+	)";
+	EVMDialectTyped const& dialect = EVMDialectTyped::instance(EVMVersion{});
+	shared_ptr<Block> result = parse(sourceText, dialect, reporter);
+	BOOST_REQUIRE(!!result);
+	BOOST_CHECK(result->debugData->astID == int64_t(7));
+	auto const& funDef = get<FunctionDefinition>(result->statements.at(0));
+	BOOST_CHECK(funDef.debugData->astID == int64_t(2));
```

Comment:
Oh, looks like there are no dedicated `int64_t` literals in C++? Apparently the standard way to get such a literal is `INT64_C(7)`. But then I guess that's not much better than a cast.